packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -412,3 +368,106 @@ class _SignUpPageState extends State<SignUpPage> with TraceableClientMixin {
+  final ValueChanged<bool> onCheckboxChanged;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    return InkWell(
+      onTap: () {
+        onCheckboxChanged(!agree);
+      },
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          IntrinsicHeight(
+            child: Row(
+              children: <Widget>[
+                IgnorePointer(
+                  ignoring: true,
+                  child: Checkbox(
+                    value: agree,
+                    fillColor: MaterialStateProperty.resolveWith(
+                      checkboxColorResolver,
+                    ),
+                    onChanged: (_) {},
+                  ),
+                ),
+                const SizedBox(width: 20.0),
+                Expanded(
+                  child: RichText(
+                    text: TextSpan(
+                      children: <InlineSpan>[
+                        TextSpan(
+                          // additional space needed because of the next text span
+                          text: '${appLocalizations.sign_up_page_agree_text} ',
+                          style: theme.textTheme.bodyText2?.copyWith(
+                            color: theme.colorScheme.onBackground,
+                          ),
+                        ),
+                        TextSpan(
+                          style: theme.textTheme.bodyText2?.copyWith(
+                            color: Colors.blue,
+                          ),
+                          text: appLocalizations.sign_up_page_terms_text,
+                          recognizer: TapGestureRecognizer()
+                            ..onTap = () => _onTermsClicked(appLocalizations),
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+                InkWell(
+                  onTap: () => _onTermsClicked(appLocalizations),
+                  customBorder: const CircleBorder(),
+                  child: AspectRatio(
+                    aspectRatio: 1.0,
+                    child: Icon(
+                      Icons.info,
+                      color: checkboxColorResolver(
+                        <MaterialState>{MaterialState.selected},
+                      ),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+          Offstage(
+            offstage: !disagree,
+            child: Text(
+              appLocalizations.sign_up_page_agree_error_invalid,
+              style: TextStyle(color: theme.errorColor),
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  Future<void> _onTermsClicked(AppLocalizations appLocalizations) async {
+    final String url = appLocalizations.sign_up_page_agree_url;
+
+    if (await canLaunchUrl(Uri.parse(url))) {
```

Comment:
I guess we are good with the try catch here, I remember last time i fixed a bug that required me to do something which got fixed after the try catch , canLunchUrl seems to be buggy sometimes 
#3182

Also we have a separate class `UrlLaunchHelper`